src/core_plugins/kibana/public/visualize/listing/visualize_listing_table.js

```diff
@@ -0,0 +1,261 @@
+        isSortAscending: this.sortableProperties.isAscendingByName('type'),
+      },
+    ];
+  }
+
+  renderRowCells(item) {
+    return [
+      <a className="kuiLink" href={this.getUrlForItem(item)}>
+        {item.title}
+      </a>,
+      <span className="kuiStatusText">
+        {this.renderItemTypeIcon(item)}
+        {item.type.title}
+      </span>
+    ];
+  }
+
+  createRows() {
+    return this.state.pageOfItems.map(item => ({
+      id: item.id,
+      cells: this.renderRowCells(item)
+    }));
+  }
+
+  closeModal = () => {
+    this.setState({ showDeleteModal: false });
+  };
+
+  renderConfirmDeleteModal() {
+    return (
+      <KuiModalOverlay>
+        <KuiConfirmModal
+          message="Are you sure you want to delete the selected visualizations? This action is irreversible!"
+          title="Warning"
+          onCancel={this.closeModal}
+          onConfirm={this.deleteSelectedItems}
+          cancelButtonText="Cancel"
+          confirmButtonText="Delete"
+        />
+      </KuiModalOverlay>
+    );
+  }
+
+  onDelete = () => {
+    this.setState({ showDeleteModal: true });
+  };
+
+  deleteSelectedItems = () => {
+    this.props.deleteSelectedItems(this.state.selectedRowIds)
+      .then(() => this.fetchItems(this.state.filter))
+      .catch(() => {})
+      .then(() => this.deselectAll())
+      .then(() => this.closeModal());
+  };
+
+  onItemSelectionChanged = (newSelectedIds) => {
+    this.setState({ selectedRowIds: newSelectedIds });
+  };
+
+  onCreate() {
+    window.location = '#/visualize/new';
+  }
+
+  renderToolBarActions() {
+    return this.state.selectedRowIds.length > 0 ?
+      <KuiListingTableDeleteButton onDelete={this.onDelete} aria-label="Delete selected visualizations"/> :
+      <KuiListingTableCreateButton onCreate={this.onCreate} aria-label="Create new visualization"/>;
+  }
+
+  renderPager() {
+    return (
+      <KuiPager
+        startNumber={this.state.pageStartNumber}
+        hasNextPage={this.pager.hasNextPage}
+        hasPreviousPage={this.pager.hasPreviousPage}
+        endNumber={this.pager.endItem}
+        totalItems={this.items.length}
+        onNextPage={this.onNextPage}
+        onPreviousPage={this.onPreviousPage}
+      />
+    );
+  }
+
+  renderPrompt() {
+    if (this.state.isFetchingItems) {
+      return <KuiListingTableLoadingPrompt />;
+    }
+
+    if (this.items.length === 0) {
+      if (this.state.filter) {
+        return <KuiListingTableNoMatchesPrompt />;
+      }
+
+      return <NoVisualizationsPrompt />;
+    }
+
+    return null;
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.showDeleteModal && this.renderConfirmDeleteModal()}
+        <KuiListingTable
+          pager={this.renderPager()}
+          toolBarActions={this.renderToolBarActions()}
+          selectedRowIds={this.state.selectedRowIds}
+          rows={this.createRows()}
+          header={this.renderHeader()}
+          onFilter={this.fetchItems}
```

Comment:
unfortunately it does require a re-fetch because we don't grab all the items, we only grab the first 100.  Eventually we'll implement paging, which will still require a re-fetch.

Reply:
I'm guessing that's for performance issues? Have we explored doing a scroll with the ES search so we can continue searching in the background while results are available? It might not be a big deal, but it seems like we could show the initial 100, then quickly load the rest behind the scene?

Reply:
One of our top notch community members explored this but ran into some complications with sorting and our index not having keyword fields for type and description.  Read through https://github.com/elastic/kibana/issues/11415 to get the background.  Something we want to do, but haven't had time to dig in deeper.  We closed the issue because as a workaround we have a configurable setting to how many items can be retrieved.

Reply:
Thanks! Makes sense!